javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,6 +56,24 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
```

Comment:
Is always clear that Iterable isn't null? We could also do the check here and not in Vector (but don't necessarily have to...)